modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/DiscoveryClientSocketTest.java

```diff
@@ -0,0 +1,175 @@
+                try {
+                    fakeTcpDiscoverySpi.configureSocketOptions(connection);
+
+                    InputStream in = connection.getInputStream();
+                    OutputStream out = connection.getOutputStream();
+
+                    int c;
+                    while ((c = in.read()) != -1)
+                        out.write(c);
+
+                    out.close();
+                    in.close();
+                    connection.close();
+                }
+                catch (Exception e) {
+                    U.closeQuiet(connection);
+
+                    System.out.println("Ex: " + e.getMessage() + " (Socket closed)");
+
+                    break;
+                }
+            }
+
+            clientFut.get();
+        }
+    }
+
+    /**
+     * Test starts ssl client socket and writes data until socket's write blocking. When the socket is blocking on write
+     * tries to close it.
+     */
+    public void startSslClient() {
+        try (Socket clientSocket = sslSockFactory1.createSocket(HOST, PORT_TO_LNSR)) {
+            System.out.println("Client started.");
+
+            fakeTcpDiscoverySpi.configureSocketOptions(clientSocket);
+
+            long handshakeStartTime = System.currentTimeMillis();
+
+            //need to send message in order to ssl handshake passed.
+            clientSocket.getOutputStream().write(IgniteUtils.IGNITE_HEADER);
+
+            byte[] buf = new byte[4];
+            int read = 0;
+
+            while (read < buf.length) {
+                int r = clientSocket.getInputStream().read(buf, read, buf.length - read);
+
+                if (r >= 0)
+                    read += r;
+                else
+                    fail("Failed to read from socket.");
+            }
+
+            assertEquals("Handshake did not pass, readed bytes: " + read, read, buf.length);
+
+            long handshakeInterval = System.currentTimeMillis() - handshakeStartTime;
+
+            System.out.println("Echo: " + U.bytesToInt(buf, 0) + " handshake time: " + handshakeInterval + "ms");
+
+            int iter = 0;
+
+            try {
+                while (true) {
+                    iter++;
+
+                    IgniteInternalFuture writeFut = GridTestUtils.runAsync(() -> {
+                        try {
+                            clientSocket.getOutputStream().write(new byte[4 * 1024]);
+                        }
+                        catch (IOException e) {
+                            assertEquals("Socket closed", e.getMessage());
+                        }
+                    });
+
+                    writeFut.get(10 * handshakeInterval);
+                }
+            }
+            catch (IgniteFutureTimeoutCheckedException e) {
+                System.out.println("Socket stuck on write, when passed " + (iter * 4) + "KB through itself.");
+            }
+
+            System.out.println("Try to close a socket.");
+
+            clientSocket.close();
```

Comment:
No need

Reply:
You are right, not need do it explicitly.